test/dispatcher/libdispatcher.c

```diff
@@ -0,0 +1,69 @@
+/*
+This file is part of GOTCHA.  For copyright information see the COPYRIGHT
+file in the top level directory, or at
+https://github.com/LLNL/gotcha/blob/master/COPYRIGHT
+This program is free software; you can redistribute it and/or modify it under
+the terms of the GNU Lesser General Public License (as published by the Free
+Software Foundation) version 2.1 dated February 1999.  This program is
+distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the IMPLIED WARRANTY OF MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the terms and conditions of the GNU Lesser General Public License
+for more details.  You should have received a copy of the GNU Lesser General
+Public License along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#define _GNU_SOURCE
+#include <dlfcn.h>
+#include <math.h>
+#include <pthread.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+int foo(void);
+int bar(void);
+
+static void* impl_lib;
+static int (*impl_foo)(void);
+static int (*impl_bar)(void);
+
+static pthread_once_t init_once = PTHREAD_ONCE_INIT;
+void dispatch_init(void) {
+  fprintf(stderr, "Ed dispatch_init()\n");
+
+  impl_lib = dlopen("libimpl.so", RTLD_NOW);
+  impl_foo = dlsym(impl_lib, "foo");
+  impl_bar = dlsym(impl_lib, "bar");
```

Comment:
please add some error checking here. when running without `LD_LIBRARY_PATH` set, the test hangs again bacause the subsequent `dlsym` calls return the symbols from the dispatcher library again

```suggestion
  impl_lib = dlopen("libimpl.so", RTLD_NOW);
  assert(impl_lib);
  impl_foo = dlsym(impl_lib, "foo");
  assert(impl_foo);
  impl_bar = dlsym(impl_lib, "bar");
  assert(impl_bar);
```

Reply:
FIXED